reconcile/openshift_base.py

```diff
@@ -621,8 +621,58 @@ def realize_data(
     return list(itertools.chain.from_iterable(results))
 
 
+def _validate_resources_used_exist(
+    ri: ResourceInventory,
+    oc: OC,
+    spec: Dict[str, Any],
+    cluster: str,
+    namespace: str,
+    kind: str,
+    name: str,
+    used_kind: str,
+):
+    # used_resources = oc.get_resources_used_in_pod_spec(spec, used_kind)
+    # TODO: use previous line instead of the next
+    used_resources: Dict[str, Set[str]] = {}
+    for used_name, used_keys in used_resources.items():
+        # perhaps used resource is deployed together with the using resource?
+        resource = ri.get_desired(cluster, namespace, used_kind, used_name)
+        if not resource:
+            # no. perhaps used resource exists in the namespace?
+            resource = oc.get(
+                namespace, used_kind, name=used_name, allow_not_found=True
+            )
+        err_base = f"[{kind}/{name}] {used_kind} {used_name}"
+        if not resource:
+            # no. where is used resource hiding? we can't find it anywhere
+            logging.error(f"{err_base} does not exist")
+            ri.register_error()
+            continue
+        # here it is! let's make sure it has all the required keys
+        for k in used_keys:
+            if k not in resource["data"]:
+                logging.error(f"{err_base} does not contain key: {k}")
+                ri.register_error()
+                continue
+
+
+def validate_planned_data(ri: ResourceInventory, oc_map: OC_Map):
+    for cluster, namespace, kind, data in ri:
+        oc = oc_map.get(cluster)
+
+        for name, d_item in data["desired"].items():
+            if kind in ("Deployment", "DeploymentConfig"):
```

Comment:
This looks worth moving up and into a `continue`, right?

```python
if kind not in ("Deployment", "DeploymentConfig"):
    continue

for name, d_item in data["desired"].items():
    ...
```

Reply:
i'd prefer leaving things the way they are to allow room for future enhancements for additional kinds. does that make sense?

Reply:
Up to you. It's a minor nit, indeed.